lib/SIL/IR/SILDeclRef.cpp

```diff
@@ -696,6 +696,28 @@ std::string SILDeclRef::mangle(ManglingKind MKind) const {
             return SS.str();
           }
           return namedClangDecl->getName().str();
+        } else if (auto objcDecl = dyn_cast<clang::ObjCMethodDecl>(clangDecl)) {
+          if (objcDecl->isDirectMethod()) {
+            // FIXME: Copied from `clang::CGObjCCommonMac:GetNameForMethod()`.
```

Comment:
Okay, no.  We should have a way to get this declaration that doesn't duplicate code across projects like this.  If that means that the mangling logic for direct methods in Clang needs to be moved into clang/AST, so be it.

Reply:
I've submitted https://reviews.llvm.org/D88329 which makes `MangleContext::mangleObjCMethodNameWithoutSize()` available to use here.

Reply:
This has been committed to trunk; you might need to shepherd it into the stable branch.